tests/framework/utils.py

```diff
@@ -60,29 +61,116 @@ class CONST:
 
 CONST = CONST()
 
+class ClusterManager(ABC):
+    EXTERNAL_CLUSTER = "EXTERNAL_CLUSTER"
 
-class KubernetesClusterManager:
+    @abstractmethod
+    def initialize_cluster(self, kind_config=None) -> None:
+        pass
+
+    @abstractmethod
+    def cleanup(self) -> None:
+        pass
+
+    @abstractmethod
+    def upload_image():
+        pass
+
+    @abstractmethod
+    def check_cluster_exist(self) -> bool:
+        pass
+    
+    @classmethod
+    def instance(cls):
+        if cls.EXTERNAL_CLUSTER in os.environ:
+            return ExternalClusterManager()
+        else:
+            return KindClusterManager()
+
+class ExternalClusterManager(ClusterManager):
+    def __init__(self) -> None:
+        config.load_kube_config()
+        self.k8s_client_dict = {}
+        self.k8s_client_dict.update(
+            {
+                CONST.K8S_V1_CLIENT_KEY: client.CoreV1Api(),
+                CONST.K8S_CR_CLIENT_KEY: client.CustomObjectsApi(),
+            }
+        )
+        self.cleanup_timeout = 120
+    
+    def cleanup(self, namespace = "default") -> None:
+        self.__delete_all_crs("ray.io", "v1alpha1", namespace, "rayservices")
```

Comment:
The custom resource lifecycles are better controlled by CREvent ([link](https://github.com/ray-project/kuberay/blob/2292e61b9155d0cd94d11ef1af987f6ff3559eb7/tests/framework/prototype.py#L160-L212)). In my opinion, the cleanup at the cluster level is not necessary. We can remove this function.